megamock/megamocks.py

```diff
@@ -414,8 +415,19 @@ def _set_attr_annotations_check(self, key: str, value: Any) -> None:
             def raise_type_error() -> None:
                 raise TypeError(f"{value!r} is not an instance of {allowed_values}")
 
-            if not isinstance(value, allowed_values):
-                raise_type_error()
+            # This is pretty basic but should handle most cases
+            # Perhaps in the future this could handle things perfectly
+            try:
+                is_allowed_type = isinstance(value, allowed_values)
+            except TypeError:
+                origin = get_origin(allowed_values)
+                if not origin:
+                    raise
```

Comment:
I'm not sure this is possible but if it does happen somehow raising makes the most sense